subsys/net/ip/net_core.c

```diff
@@ -371,10 +367,8 @@ static void net_queue_rx(struct net_if *iface, struct net_pkt *pkt)
 	k_work_init(net_pkt_work(pkt), process_rx_packet);
 
 #if defined(CONFIG_NET_STATISTICS)
-	pkt->total_pkt_len = net_pkt_get_len(pkt);
-
 	net_stats_update_tc_recv_pkt(iface, tc);
-	net_stats_update_tc_recv_bytes(iface, tc, pkt->total_pkt_len);
+	net_stats_update_tc_recv_bytes(iface, tc, net_pkt_get_len(pkt));
```

Comment:
We need to figure out better way to do this at some point. The `net_pkt_get_len()` will traverse the buffer list and sum the total length. This is now done multiple times in worst case if CONFIG_NET_STATISTICS is enabled, first time in `net_queue_rx()` and then again in `net_rx()`

Reply:
Actually I did have a way to improve that in my prototype back then, basically having a length attribute in net_pkt, that was affected by net_pkt_writ/skip/memset/pull if overwrite was false.
But I wonder if that's necessary, on ethernet frames if we take a net_buf size of 128 bytes we would have at worst 11 net_buf. Of course if we call net_pkt_get_len() a lot, then that's another issue. It's a trade off I guess, whether we want to save 2bytes from net_pkt of some cpu. A bit of benchmarking wouldn't hurt to figure out, let's see this in a near future.